pkg/asset/manifests/ingress.go

```diff
@@ -78,6 +78,24 @@ func (ing *Ingress) Generate(dependencies asset.Parents) error {
 }
 
 func (ing *Ingress) generateClusterConfig(config *types.InstallConfig) ([]byte, error) {
+	controlPlaneTopology, infrastructureTopology := determineTopologies(config)
+
+	isSingleNode := controlPlaneTopology == configv1.SingleReplicaTopologyMode &&
+		infrastructureTopology == configv1.SingleReplicaTopologyMode
```

Comment:
Why is the infrastructure topology a consideration? Shouldn't the logic be that if there is a single control-plane node, then the default placement is on the control plane?

From https://github.com/openshift/enhancements/blob/master/enhancements/single-node/single-node-openshift-with-workers.md#implementation-detailsnotesconstraints,
```
The installer will detect situations in which it's unlikely the user will want to set up a load-balancer. For now, those situations only include installation of single control-plane node cluster deployments on the "none" platform.
```

Reply:
(2) has the advantage of being consistent - a single control plane node cluster by default will always have ingress go through the control plane node, regardless of how many workers it has, and never requires the user to provision a load balancer because the SNO acts like one.

(1) has the advantage of leveraging the user's hardware better by default, but complicates things because the user has to provision a load balancer for ingress (but not one for the API) - which is inconsistent and would probably require good documentation to get the idea across

Reply:
I vote for consistency here